core/src/main/java/io/smallrye/openapi/runtime/util/KotlinUtil.java

```diff
@@ -0,0 +1,53 @@
+package io.smallrye.openapi.runtime.util;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.FieldInfo;
+import org.jboss.jandex.MethodInfo;
+
+/**
+ * Some utility methods for working with Kotlin types.
+ *
+ * @author Nicklas Jensen {@literal <nillerr@gmail.com>}
+ */
+class KotlinUtil {
+    /**
+     * Constructor.
+     */
+    private KotlinUtil() {
+    }
+
+    private static String getGetterName(FieldInfo field) {
+        return "get" + field.name().substring(0, 1).toUpperCase() + field.name().substring(1);
+    }
+
+    private static String getSyntheticPropertyAnnotationsMethodName(FieldInfo field) {
+        // Kotlin generates a synthetic method for each property declaring any `AnnotationTarget.PROPERTY` annotations.
+        // The name of the property is the getter-name of the field followed by `$annotations`.
+        return getGetterName(field) + "$annotations";
+    }
+
+    private static Optional<MethodInfo> getSyntheticPropertyAnnotationsMethod(FieldInfo field) {
+        var methodName = getSyntheticPropertyAnnotationsMethodName(field);
+        var clazz = field.declaringClass();
+        return clazz.methods().stream()
+                .filter(m -> Modifiers.isSynthetic(m.flags()) && methodName.equals(m.name()))
```

Comment:
Use `MethodInfo#isSynthetic` and the `Modifiers` util class can be removed too.

```suggestion
                .filter(m -> m.isSynthetic() && methodName.equals(m.name()))
```